pennylane/qchem/vibrational/taylor_ham.py

```diff
@@ -0,0 +1,594 @@
+        alphas_arr = np.zeros((num_modes, num_modes))
+        for m in range(num_modes):
+            alphas_arr[m, m] = freqs[m]
+
+    kin_ham = BoseSentence({})
+    for m1 in range(num_modes):
+        pm1 = _position_to_boson(m1, "p")
+        for m2 in range(num_modes):
+            pm2 = _position_to_boson(m2, "p")
+            kin_ham += (0.5 * alphas_arr[m1, m2]) * (pm1 * pm2).normal_order()
+
+    return kin_ham.normal_order()
+
+
+def _taylor_harmonic(taylor_coeffs_array, freqs):
+    """Build harmonic term of Taylor form bosonic observable from provided coefficients, see first
+    term of `Eq. 4 and Eq. 7 <https://arxiv.org/pdf/1703.09313>`_.
+
+    Args:
+        taylor_coeffs_array (list(float)): the coeffs of the Taylor expansion
+        freqs (list(float)): vibrational frequencies
+
+    Returns:
+        BoseSentence: harmonic term of the Taylor hamiltonian for given coeffs
+    """
+    taylor_1D = taylor_coeffs_array[0]
+    num_modes, _ = np.shape(taylor_1D)
+    harm_pot = BoseSentence({})
+    for mode in range(num_modes):
+        bosonized_qm2 = (
+            _position_to_boson(mode, "q") * _position_to_boson(mode, "q")
+        ).normal_order()
+        harm_pot += bosonized_qm2 * freqs[mode] * 0.5
+
+    return harm_pot.normal_order()
+
+
+def taylor_bosonic(taylor_coeffs_array, freqs, is_local=True, uloc=None):
+    """Return Taylor bosonic vibrational Hamiltonian.
+
+     The construction of the Hamiltonian is based on Eqs. 4-7 of `arXiv:1703.09313 <https://arxiv.org/abs/1703.09313>`_.
+
+    Args:
+        taylor_coeffs (list(float)): the coefficients of the Hamiltonian
+        freqs (list(float)): the harmonic frequencies in reciprocal centimetre
+        is_localal (bool): Flag whether the vibrational modes are localized. Default is ``True``.
+        uloc (list(float)): localization matrix indicating the relationship between original and
+            localized modes
+
+    Returns:
+        BoseSentence: Taylor bosonic hamiltonian
+    """
+    if is_local:
+        start_deg = 2
+    else:
+        start_deg = 3
+
+    harm_pot = _taylor_harmonic(taylor_coeffs_array, freqs)
+    ham = _taylor_anharmonic(taylor_coeffs_array, start_deg) + harm_pot
+    kin_ham = _taylor_kinetic(taylor_coeffs_array, freqs, is_local, uloc)
+    ham += kin_ham
+    return ham.normal_order()
+
+
+def taylor_hamiltonian(
+    pes, max_deg=4, min_deg=3, mapping="binary", n_states=2, ps=False, wire_map=None, tol=None
+):
+    """Return Taylor vibrational Hamiltonian.
+
+    Args:
+        pes (VibrationalPES): object containing the vibrational potential energy surface data
+        max_deg (int): maximum degree of Taylor form polynomial
+        min_deg (int): minimum degree of Taylor form polynomial
+        mapping (str): Mapping used to map to qubit basis. Default is `"binary"`.
+        n_states(int): maximum number of allowed bosonic states
+        ps (bool): Whether to return the result as a PauliSentence instead of an
```

Comment:
```suggestion
        ps (bool): Flag to return the result as a PauliSentence instead of an
```

Reply:
Please correct everywhere bool is being used.

Reply:
Think I copied this from bosonic mappings.